public/app/core/components/AppChrome/AppChrome.tsx

```diff
@@ -53,6 +54,18 @@ export function AppChrome({ children }: Props) {
     chrome.setMegaMenuOpen(!state.megaMenuOpen);
   };
 
+  const shouldShowReturnToPrevious = () => {
```

Comment:
mentioned this in 1-1, but i think this logic can be split out.

in my mind, you have something like:
- `const shouldShowReturnToPrevious = config.featureToggles.returnToPrevious && state.returnToPrevious`
- some completely separate logic to clear the returnToPrevious state when the location matches the existing returnToPrevious href. maybe that would be better as an effect?:
  ```tsx
  useEffect(() => {
    if (state.returnToPrevious && location.pathname === state.returnToPrevious.href) {
      clearReturnToPrevious();
    }
  }, [location.pathname])
  ```